example/analysis_options.yaml

```diff
@@ -27,4 +27,18 @@ linter:
     throw_in_finally: true
     type_annotate_public_apis: true
     unnecessary_statements: true
-    use_named_constants: true
\ No newline at end of file
+    use_named_constants: true
+    use_super_parameters: true
+    cast_nullable_to_non_nullable: true
+    only_throw_errors: true
+    sort_unnamed_constructors_first: true
+    prefer_asserts_with_message: true
+    missing_whitespace_between_adjacent_strings: true
+    avoid_multiple_declarations_per_line: true
+    noop_primitive_operations: true
+    avoid_void_async: true
+    avoid_types_on_closure_parameters: true
+    unnecessary_null_checks: true
+    unnecessary_parenthesis: true
+    use_if_null_to_convert_nulls_to_bools: true
+    matching_super_parameters: true
```

Comment:
Shouldn't we add `flutter_lints` as a dev_dependency to the example instead? This would avoid to maintain the rule set 🤔

Reply:
We are already using `flutter_lints` in the example project. See in line 1:
```yaml
include: package:flutter_lints/flutter.yaml
```
There are additional linters specified in the `flutter_map/analysis_options.yaml` file that I thought should match the example project at `flutter_map/example/analysis_options.yaml`.

The only rules I **haven't** copied over are
- `avoid_redundant_argument_values` because it can make sence to specify the default paramters in the example app as showcase and for explainational purpose.
- `prefer_single_quotes` since the project currently uses both `"` and `'`. I could change this tho for more concistency.

But I agree that it would be nice to use some trick here do match the rules with the package if somehow possible.

Reply:
Oh my bad you're right!
We might want to update our version of flutter_lints because from what I see, the latest release exposes some new rules such as `use_super_parameters`, so we could remove it from our `analysis_options`.

Reply:
Agree, completely missed that it has a new major version. 😅 

Btw see my changes to `example/analysis_options.yaml`. We are now matching the package lint rules automatically by using:
```yaml
include: ./../analysis_options.yaml
```